File: FlickrApp/ViewControllers/UIViewController+LoadingView.m
//
//  UIViewController+LoadingView.m
//  FlickrApp
//
//  Created by aSqar on 26.11.2017.
//  Copyright © 2017 Askar Bakirov. All rights reserved.
//

#import "UIViewController+LoadingView.h"
#import "DejalActivityView.h"

@implementation UIViewController(LoadingView)

- (void) showLoadingView: (NSString *) msg
{
    if ([DejalBezelActivityView currentActivityView] != nil)
    {
        [DejalBezelActivityView currentActivityView].activityLabel.text = msg;
    } else {
        [DejalBezelActivityView activityViewForView: self.view withLabel: msg width:250.0f];
    }
}

- (void) hideLoadingView
{
    [DejalBezelActivityView removeViewAnimated: YES];
}

@end

File: FlickrApp/ViewControllers/UIViewController+LoadingView.swift
//
//  UIViewController+LoadingView.swift
//  FlickrApp
//
//  Created by aSqar on 26.11.2017.
//  Copyright © 2017 Askar Bakirov. All rights reserved.
//

import UIKit
import KVNProgress

extension UIViewController {

    func showLoadingView(msg:String!) {
        KVNProgress.show(withStatus: msg)
    }

    func hideLoadingView() {
        KVNProgress.dismiss()
    }
}
